diff --git a/net/netmon/netmon.go b/net/netmon/netmon.go
--- a/net/netmon/netmon.go
+++ b/net/netmon/netmon.go
@@ -372,6 +372,9 @@ func filterRoutableIPs(addrs []netip.Prefix) []netip.Prefix {
 // Use RegisterChangeCallback to get notified of network changes.
 func New(bus *eventbus.Bus, logf logger.Logf) (*Monitor, error) {
 	logf = logger.WithPrefix(logf, "monitor: ")
+	if runtime.GOOS == "android" {
+		return newStatic(bus, true), nil
+	}
 	m := &Monitor{
 		logf:     logf,
 		b:        bus.Client("netmon"),
@@ -401,9 +404,20 @@ func New(bus *eventbus.Bus, logf logger.Logf) (*Monitor, error) {
 // but doesn't actually monitor for changes. It should only be used in tests
 // and situations like cleanups or short-lived CLI programs.
 func NewStatic() *Monitor {
+	return newStatic(nil, false)
+}
+
+func newStatic(bus *eventbus.Bus, assumeNetwork bool) *Monitor {
 	m := &Monitor{static: true}
+	if bus != nil {
+		m.b = bus.Client("netmon")
+		m.changed = eventbus.Publish[ChangeDelta](m.b)
+	}
 	if st, err := m.interfaceStateUncached(); err == nil {
 		m.ifState = st
 	}
+	if m.ifState == nil && assumeNetwork {
+		m.ifState = &State{HaveV4: true}
+	}
 	return m
 }
